src/element/mod.rs

```diff
@@ -253,6 +321,31 @@ impl IonEq for Element {
     }
 }
 
+// Ordering is done as follows:
+// 1. Ion type -- because it is the cheapest comparison
```

Comment:
I have a stronger rationale than just expense--actually ordering the Ion types together makes logical sense for the natural ordering we're defining here.

Reply:
One could also make a case for annotations being the first thing to sort by. (I'm not going to argue that here, but I understand that as a point of view.) I can update the rationale.